pkg/middleware/middleware_test.go

```diff
@@ -0,0 +1,94 @@
+package middleware_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/server"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAuthMiddleware(t *testing.T) {
+
+	// create a TokenGenerator and a key
+	tokengenerator := testutils.NewTokenManager()
+	kid0 := uuid.NewV4().String()
+	_, err := tokengenerator.AddPrivateKey(kid0)
+	require.NoError(t, err)
+
+	// create some test tokens
+	identity0 := testutils.Identity {
+		ID: uuid.NewV4(),
+		Username: uuid.NewV4().String(),
+	}
+	emailClaim0 := testutils.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
+	tokenValid, err := tokengenerator.GenerateSignedToken(identity0, kid0, emailClaim0)
+	require.NoError(t, err)
+	tokenInvalidNoEmail, err := tokengenerator.GenerateSignedToken(identity0, kid0)
+	require.NoError(t, err)
+	tokenInvalidGarbage := uuid.NewV4().String()
+
+	// start key service
+	keysEndpointURL := tokengenerator.NewKeyServer().URL
+
+	// create server
+	srv, err := server.New("")
+	require.NoError(t, err)	
+
+	// set the key service url in the config
+	os.Setenv(configuration.EnvPrefix+"_"+"AUTH_CLIENT_PUBLIC_KEYS_URL", keysEndpointURL)
+	assert.Equal(t, keysEndpointURL, srv.Config().GetAuthClientPublicKeysURL(), "key url not set correctly")
+	os.Setenv(configuration.EnvPrefix+"_"+"TESTINGMODE", "true")
+	assert.True(t, srv.Config().IsTestingMode(), "testing mode not set correctly")
+	
+	// Setting up the routes.
+	err = srv.SetupRoutes()
+	require.NoError(t, err)
+
+	// Check that there are routes registered.
+	routes := srv.GetRegisteredRoutes()
+	require.NotEmpty(t, routes)
+
+	// Check that Engine() returns the router object.
+	require.NotNil(t, srv.Engine())
+
+	// do some requests
+	var authtests = []struct {
+		name    		string
+		urlPath 		string
+		method  		string
+		tokenHeader	string
```

Comment:
Looks like a formatting issue. Do you have `go fmt` setup in your IDE?

Reply:
Fixed c85005c.